src/app/common/transactions/bitcoin/coinselect/local-coin-selection.ts

```diff
@@ -96,3 +103,103 @@ export function determineUtxosForSpend({
     fee,
   };
 }
+
+export interface DetermineUtxosForSpendArgsMultipleRecipients {
+  amount: number;
+  feeRate: number;
+  recipients: RpcSendTransferRecipient[];
+  utxos: UtxoResponseItem[];
+}
+
+interface DetermineUtxosForSpendAllArgsMultipleRecipients {
+  feeRate: number;
+  recipients: RpcSendTransferRecipient[];
+  utxos: UtxoResponseItem[];
+}
+
+export function determineUtxosForSpendAllMultipleRecipients({
+  feeRate,
+  recipients,
+  utxos,
+}: DetermineUtxosForSpendAllArgsMultipleRecipients) {
+  recipients.forEach(recipient => {
+    if (!validate(recipient.address))
+      throw new Error('Cannot calculate spend of invalid address type');
+  });
+  const filteredUtxos = filterUneconomicalUtxosMultipleRecipients({ utxos, feeRate, recipients });
+
+  const sizeInfo = getSizeInfoMultipleRecipients({
+    inputLength: filteredUtxos.length,
+    isSendMax: true,
+    recipients,
+  });
+
+  // Fee has already been deducted from the amount with send all
+  const outputs = recipients.map(({ address, amount }) => ({ value: BigInt(amount), address }));
+
+  const fee = Math.ceil(sizeInfo.txVBytes * feeRate);
```

Comment:
use BigNumber for math

Reply:
most of this fn is copy-pasted, I can refactor math there later?